src/python/pants/init/specs_calculator.py

```diff
@@ -75,56 +66,47 @@ def create(
     exclude_patterns: Optional[Iterable[str]] = None,
     tags: Optional[Iterable[str]] = None,
   ) -> Specs:
-    # Determine the literal specs.
     specs = cls.parse_specs(
       raw_specs=options.specs,
       build_root=build_root,
       exclude_patterns=exclude_patterns,
       tags=tags,
     )
 
-    # Determine `Changed` arguments directly from options to support pre-`Subsystem`
-    # initialization paths.
-    changed_options = options.for_scope('changed')
-    changed_request = ChangedRequest.from_options(changed_options)
-
-    # Determine the `--owner-of=` arguments provided from the global options
+    changed_options = ChangedOptions.from_options(options.for_scope('changed'))
     owned_files = options.for_global_scope().owner_of
 
     logger.debug('specs are: %s', specs)
-    logger.debug('changed_request is: %s', changed_request)
+    logger.debug('changed_options are: %s', changed_options)
     logger.debug('owned_files are: %s', owned_files)
     targets_specified = sum(
       1 for item
-      in (changed_request.is_actionable(), owned_files, specs.provided_specs.dependencies)
+      in (changed_options.is_actionable(), owned_files, specs.provided_specs.dependencies)
       if item
     )
 
     if targets_specified > 1:
       # We've been provided more than one of: a change request, an owner request, or specs.
       raise InvalidSpecConstraint(
         'Multiple target selection methods provided. Please use only one of '
-        '--changed-*, --owner-of, address specs, or filesystem specs.'
+        '`--changed-*`, `--owner-of`, address specs, or filesystem specs.'
       )
 
-    if changed_request.is_actionable():
+    if changed_options.is_actionable():
       scm = get_scm()
       if not scm:
         raise InvalidSpecConstraint(
-          'The --changed-* options are not available without a recognized SCM (usually git).'
+          'The `--changed-*` options are not available without a recognized SCM (usually git).'
         )
-      changed_files = cls.changed_files(
-          scm,
-          changes_since=changed_request.changes_since,
-          diffspec=changed_request.diffspec)
```

Comment:
By moving `changed_files` to belong to `ChangedOptions`, we no longer need to pass `changes_since` and `diffspec`. Now, it's simply `changed_options.changed_files(scm=scm)`.